packages/teams-js/test/public/appNotification.spec.ts

```diff
@@ -0,0 +1,215 @@
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: {} });
+          expect.assertions(1);
+          try {
+            appNotification.displayInAppNotification(displayNotificationParam);
+          } catch (e) {
+            expect(e).toEqual(errorNotSupportedOnPlatform);
+          }
+        });
+
+        it('displayInAppNotification call with successful result', async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { appNotification: {} } });
+          const promise = appNotification.displayInAppNotification(displayNotificationParam);
+          const message = utils.findMessageByFunc('appNotification.displayNotification');
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toEqual(displayNotificationParamForAppHost);
+          //representation of what hubsdk sends to us in the teamsJS
+          const callbackId = message?.id;
+          utils.respondToFramelessMessage({
+            data: {
+              id: callbackId,
+              args: [undefined, null],
+            },
+          } as DOMMessageEvent);
+          await expect(promise).resolves.toBe(null);
+        });
+
+        it('displayInNotification call should not be successful with invalid title length', async () => {
+          const maxTitleLength = 75;
+          await utils.initializeWithContext(FrameContexts.content);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { appNotification: {} } });
+          const displayNotificationParam: appNotification.NotificationDisplayParam = {
+            title:
+              'Update: The upcoming maintenance scheduled for tomorrow at 10:00 AM has been extended to include additional server upgrades. Please plan accordingly.',
+            content: `
+                Dear valued users,
+                We would like to inform you that scheduled maintenance will take place on our Microsoft platform tomorrow, August 1st, 2023, starting at 10:00 AM UTC. During this time, Microsoft services will be temporarily unavailable as we perform necessary upgrades to enhance performance and security
+                Expected Downtime: Approximately 4 hours.
+                We apologize for any inconvenience this may cause and assure you that our team at Microsoft is working diligently to complete the maintenance as quickly as possible.`,
+            displayDurationInSeconds: 30,
+            notificationActionUrl: new URL('http://www.example.com'),
+          };
+
+          expect(() => appNotification.displayInAppNotification(displayNotificationParam)).toThrowError(
+            new Error(
+              `Invalid notification title length: Maximum title length ${maxTitleLength}, title length supplied ${displayNotificationParam.title.length}`,
```

Comment:
>Invalid notification title length [](http://example.com/codeflow?start=15&length=33)

In this test, it looks like you have both a title string that is too long and a content string that is too long. Your test is validating that the error thrown is about the title length.

Unless you are trying to guarantee that whenever there are multiple parameter violations the title is ALWAYS validated first (which I don't think you are trying to guarantee), then you probably don't want to specify both an invalid title AND invalid content. Otherwise someone could change the order in which the title and content lengths are validated in the code, and this test would start failing, even though the behavior was still correct.